### PR TITLE
Feature01

### PR DESCRIPTION
**Pull request by hasi-b**
pull request to review

### CODE REVIEW

File: Assets/Feature01/Scripts/Bullet/GreenBullet.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GreenBullet : BulletAbstractClasses
+{
+
```
Comment (hasi-b) [inline Assets/Feature01/Scripts/Bullet/GreenBullet.cs]: Color should be a property.

File: Assets/Feature01/Scripts/Controller/Instantiation.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Instantiation : MonoBehaviour
```
Comment (hasi-b) [inline Assets/Feature01/Scripts/Controller/Instantiation.cs]: Enum class should not be a mono behaviour

File: Assets/Feature01/Scripts/Controller/PlayerController.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerController : MonoBehaviour
+{
+    Rigidbody rb;
+    [SerializeField] PlayerOnFootInput PlayerFootInput;
+    [SerializeField] PlayerInput pl;
```
Comment (hasi-b) [inline Assets/Feature01/Scripts/Controller/PlayerController.cs]: write fully qualified name for properties / variables

File: Assets/Feature01/Scripts/Controller/PlayerController.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        
+    }
+
+
+    private void FixedUpdate()
+    {
+        
+        PlayerMove();
+        PlayerLook();
+        rb.AddForce(playerMoveInput,ForceMode.Force);
```
Comment (hasi-b) [inline Assets/Feature01/Scripts/Controller/PlayerController.cs]: fully qualified name

File: Assets/Feature01/Scripts/Controller/Shoot/ShootSystem.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    //Rigidbody rbBullet;
+
+
+    private void Awake()
+    {
+        //rbBullet = bulletGameObject.GetComponent<Rigidbody>();
+        playerFootInput = GetComponent<PlayerOnFootInput>();
+    }
+
+    public  void playerShoot(Vector3 mousePosition)
```
Comment (hasi-b) [inline Assets/Feature01/Scripts/Controller/Shoot/ShootSystem.cs]: check if this is correct implementation with New Input System

File: Assets/Feature01/Scripts/Input/PlayerOnFootInput.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    {
+       MoveInput=context.ReadValue<Vector2>();
+    }
+
+   
+
+    public Vector3 getMousePosition()
+    {  
+        
+        return mainCamera.ScreenToWorldPoint(new Vector3(Input.mousePosition.x, Input.mousePosition.y, mainCamera.transform.position.y));
```
Comment (hasi-b) [inline Assets/Feature01/Scripts/Input/PlayerOnFootInput.cs]: Use new inputsystem only

**Review by hasi-b — COMMENTED** [review]
Need to update according to the comnments

File: Assets/Feature01/Scripts/AbstractClasses/CharacterAbstractClass.cs [hunk 7]
```diff
@@ -0,0 +1,8 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class CharacterAbstractClass
+{
+    public abstract void characterInstantiate();
```
Comment (Warhammer4000) [inline Assets/Feature01/Scripts/AbstractClasses/CharacterAbstractClass.cs]: Method Name will be CamelCase. 
First character will be capital. Need to follow one Standard so that everyone is on the same ground

File: Assets/Feature01/Scripts/Characters/Enemy.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Enemy : CharacterAbstractClass
+{
+    public override void characterInstantiate()
```
Comment (Warhammer4000) [inline Assets/Feature01/Scripts/Characters/Enemy.cs]: Capital

File: Assets/Feature01/Scripts/Characters/Player.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Player : CharacterAbstractClass
+{
+
+    public override void characterInstantiate()
```
Comment (Warhammer4000) [inline Assets/Feature01/Scripts/Characters/Player.cs]: Capital

File: Assets/Feature01/Scripts/Controller/MovementController.cs [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+public class MovementController : MonoBehaviour
+{
+
+
+   
+   
+    Vector3 playerMoveInput;
+
+    public void PlayerMove(Vector2 movementInput, float speed,Rigidbody _rigidbody)
+    {
```
Comment (Warhammer4000) [inline Assets/Feature01/Scripts/Controller/MovementController.cs:23]: Are you sure you wish to create movement based on rigid body? 
Open ended question. It can be something really fun.. Or something very complicated to manage.

File: Assets/Feature01/Scripts/Controller/PlayerController.cs [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+    private void Awake()
+    {
+        _rigidbody = GetComponent<Rigidbody>();
+     
+        
+    }
+
+
+    private void FixedUpdate()
```
Comment (Warhammer4000) [inline Assets/Feature01/Scripts/Controller/PlayerController.cs:42]: This is not how you should use New input system. You are basically using the same paradigm as old input system. I recommend watching a few tutorials.

How it should work is . whenever your stick is moving something is happening. Event driven. Not Update Driven.

File: Assets/Feature01/Scripts/Input/GamePadCursor.cs [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+
+
+
+
+
+
+
+    public void moveCursorPosition(Vector2 cursorMovementInput,float _cursorSpeed)
+    {
```
Comment (Warhammer4000) [inline Assets/Feature01/Scripts/Input/GamePadCursor.cs]: Why did you need this?
You should have a look at the cursor class on unity API.

File: Assets/Feature01/Scripts/Input/GamePadCursor.cs [hunk 13]
```diff
@@ -0,0 +1,47 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.LowLevel;
+using UnityEngine.InputSystem.Users;
+
+public class GamePadCursor : MonoBehaviour
```
Comment (Warhammer4000) [inline Assets/Feature01/Scripts/Input/GamePadCursor.cs]: Gamepad cursor is not needed. Just use the direction vector of the Right Axis. Shoot at the direction.

File: Assets/Feature01/Scripts/Input/PlayerOnFootInput.cs [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+    private void SetMove(UnityEngine.InputSystem.InputAction.CallbackContext context)
+    {
+
+        playerController.movementInput = context.ReadValue<Vector2>();
+      // MoveInput=context.ReadValue<Vector2>();
+    }
+
+   
+
+    public Vector3 getMousePosition()
```
Comment (Warhammer4000) [inline Assets/Feature01/Scripts/Input/PlayerOnFootInput.cs]: You probably don't need this. Not sure. Do you? Can you explain briefly.

**Review by Warhammer4000 — CHANGES_REQUESTED** [review]
Have a look

File: Assets/Feature01/Scripts/AbstractClasses/CharacterAbstractClass.cs (hunk 7 above)
Comment (hasi-b) [reply Assets/Feature01/Scripts/AbstractClasses/CharacterAbstractClass.cs]: yes bhai update kortesi

File: Assets/Feature01/Scripts/Controller/PlayerController.cs (hunk 11 above)
Comment (hasi-b) [reply Assets/Feature01/Scripts/Controller/PlayerController.cs:42]: bhai ami kiso tutorial dekhsilam,  oigula te emon fixed update er vitor  diye kora, event driven input  niye constantly check kore. 
bhai any recommendation on tutorial jekhane only event fire korle e move hocche

File: Assets/Feature01/Scripts/Input/GamePadCursor.cs (hunk 12 above)
Comment (hasi-b) [reply Assets/Feature01/Scripts/Input/GamePadCursor.cs]: eta gamepad diye jokhn first e cursor movement  implement korechilam tokhn kora, baad diye dicchi

File: Assets/Feature01/Scripts/Input/GamePadCursor.cs (hunk 13 above)
Comment (hasi-b) [reply Assets/Feature01/Scripts/Input/GamePadCursor.cs]: ji bhai baad ditesi eta

File: Assets/Feature01/Scripts/Input/PlayerOnFootInput.cs (hunk 14 above)
Comment (hasi-b) [reply Assets/Feature01/Scripts/Input/PlayerOnFootInput.cs]: yes bhai, jokhn mouse based shooting implement korsilam first e tokhn kora, pore baad deya hoy naai, comment out kore dicchi
